tests/Integration/Services/CRM/Activity/Service/TypeTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * This file is part of the bitrix24-php-sdk package.
+ *
+ * © Maksim Mesilov <mesilov.maxim@gmail.com>
+ *
+ * For the full copyright and license information, please view the MIT-LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Bitrix24\SDK\Tests\Integration\Services\CRM\Activity\Service;
+
+use Bitrix24\SDK\Core\Exceptions\BaseException;
+use Bitrix24\SDK\Core\Exceptions\TransportException;
+use Bitrix24\SDK\Services\CRM\Activity\ActivityContentType;
+use Bitrix24\SDK\Services\CRM\Activity\ActivityDirectionType;
+use Bitrix24\SDK\Services\CRM\Activity\Result\ActivityItemResult;
+use Bitrix24\SDK\Services\CRM\Activity\Service\Activity;
+use Bitrix24\SDK\Services\CRM\Activity\ActivityType;
+use Bitrix24\SDK\Services\CRM\Contact\Service\Contact;
+use Bitrix24\SDK\Services\CRM\Deal\Result\DealItemResult;
+use Bitrix24\SDK\Services\CRM\Deal\Result\DealProductRowItemResult;
+use Bitrix24\SDK\Tests\Builders\DemoDataGenerator;
+use Bitrix24\SDK\Tests\Integration\Fabric;
+use PHPUnit\Framework\Attributes\CoversFunction;
+use PHPUnit\Framework\Attributes\CoversMethod;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Typhoon\Reflection\TyphoonReflector;
+use Bitrix24\SDK\Tests\CustomAssertions\CustomBitrix24Assertions;
+use Bitrix24\SDK\Core;
+
+#[CoversClass(Type::class)]
+#[CoversMethod(Type::class, 'add')]
+#[CoversMethod(Type::class, 'delete')]
+#[CoversMethod(Type::class, 'list')]
+class TypeTest extends TestCase
+{
+    use CustomBitrix24Assertions;
+
+    private ActivityType $activityTypeService;
+
+    public function testAllSystemFieldsAnnotated(): void
+    {
+        $propListFromApi = (new Core\Fields\FieldsFilter())->filterSystemFields(array_keys($this->activityTypeService->fields()->getFieldsDescription()));
+        $this->assertBitrix24AllResultItemFieldsAnnotated($propListFromApi, ActivityItemResult::class);
+    }
+
+    public function testAllSystemFieldsHasValidTypeAnnotation():void
+    {
+        $this->assertBitrix24AllResultItemFieldsHasValidTypeAnnotation(
+            $this->activityTypeService->fields()->getFieldsDescription(),
+            ActivityItemResult::class);
+    }
+
+    /**
+     * @throws BaseException
+     * @throws TransportException
+     */
+    public function testAdd(): void
+    {
+        $activityTypeId = $this->activityTypeService->add(

```

Comment:
For test add you must check

1. is added type ok
2. read added type
3. compare some key fields